### PR TITLE
Codes types: execOverwriteCode must also be called for children

### PR DESCRIPTION
**Pull request by matthiaso**
For code types, codes for loadCodes are initially supplied by
execCreateCodes and additionally this list may be enriched by
execLoadCodes (e.g. modify existing codes or add additional codes). If
existing codes are modified execOverwriteCode should be called to
influence this modification. However, previously execOverwriteCode was
only called for the root codes but not for child codes, it must as well
be called for child codes.

297625

### CODE REVIEW

File: org.eclipse.scout.rt.shared.test/src/test/java/org/eclipse/scout/rt/shared/services/common/code/CodeTypeTest.java [hunk 1]
```diff
@@ -95,30 +96,66 @@ private static class TestCodeType extends AbstractCodeType<String, String> {
...
       return ID;
     }
 
+    public List<String> getOverwrittenCodes() {
+      return m_overwrittenCodes;
+    }
+
     @Override
     protected List<ICodeRow<String>> execLoadCodes(Class<? extends ICodeRow<String>> codeRowType) {
       List<ICodeRow<String>> codeRows = new ArrayList<>();
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared.test/src/test/java/org/eclipse/scout/rt/shared/services/common/code/CodeTypeTest.java]: Remove this row

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/services/common/code/AbstractCodeTypeWithGeneric.java [hunk 2]
```diff
@@ -443,11 +444,12 @@ protected void loadCodes() {
     // 1a add configured codes
     List<? extends CODE> createdList = interceptCreateCodes();
     if (createdList != null) {
-      for (CODE code : createdList) {
+      visit(createdList, (ICodeVisitor<CODE>) (code, treeLevel) -> {
         allCodesOrdered.add(code);
         idToCodeMap.put(code.getId(), code);
-        codeToParentCodeMap.put(code, null);
-      }
+        codeToParentCodeMap.put(code, Optional.ofNullable(code.getParentCode()).map(ICode::getId).map(idToCodeMap::get).orElse(null));
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/services/common/code/AbstractCodeTypeWithGeneric.java]: to improve readability I'll rather use the following snippet without optional/map:

``codeToParentCodeMap.put(code, code.getParentCode() != null ? idToCodeMap.get(code.getParentCode().getId()) : null);``

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/services/common/code/AbstractCodeTypeWithGeneric.java [hunk 3]
```diff
@@ -443,11 +444,12 @@ protected void loadCodes() {
     // 1a add configured codes
     List<? extends CODE> createdList = interceptCreateCodes();
     if (createdList != null) {
-      for (CODE code : createdList) {
+      visit(createdList, (ICodeVisitor<CODE>) (code, treeLevel) -> {
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/services/common/code/AbstractCodeTypeWithGeneric.java]: the following signature (without cast) would also compile and is a little bit more readable in my opinion:
`visit(createdList, (CODE code, int treeLevel) -> {`

**Review by paolobazzi — COMMENTED** [review]
Looks good, see comments for minor improvements.

**Comment by matthiaso** [comment]
Please run tests

**Comment by matthiaso** [comment]
Run tests

**Comment by matthiaso** [comment]
Run test

**Comment by avandorp** [comment]
Jenkins, please run tests

**Pull request merged**